tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/traversal/FolderTraversalServiceImpl.java

```diff
@@ -0,0 +1,143 @@
+package com.dotcms.api.traversal;
+
+import com.dotcms.model.asset.FolderView;
+import io.quarkus.arc.DefaultBean;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.concurrent.ForkJoinPool;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.context.control.ActivateRequestContext;
+import javax.inject.Inject;
+
+/**
+ * Service for traversing a file system directory and building a hierarchical tree representation of
+ * its contents. The traversal is performed using a ForkJoinPool, which allows for parallel
+ * execution of the traversal tasks.
+ */
+@DefaultBean
+@Dependent
+public class FolderTraversalServiceImpl implements FolderTraversalService {
+
+    @Inject
+    protected Retriever retriever;
+
+    /**
+     * Traverses the file system directory at the specified path and builds a hierarchical tree
+     * representation of its contents.
+     *
+     * @param path                  The path to the directory to traverse.
+     * @param depth                 The maximum depth to traverse the directory tree. If null, the
+     *                              traversal will go all the way down to the bottom of the tree.
+     * @param includeFolderPatterns The glob patterns for folders to include in the traversal.
+     * @param includeAssetPatterns  The glob patterns for assets to include in the traversal.
+     * @param excludeFolderPatterns The glob patterns for folders to exclude from the traversal.
+     * @param excludeAssetPatterns  The glob patterns for assets to exclude from the traversal.
+     * @return A TreeNode representing the directory tree rooted at the specified path.
+     */
+    @ActivateRequestContext
+    @Override
+    public TreeNode traverse(
+            final String path,
+            final Integer depth,
+            final Set<String> includeFolderPatterns,
+            final Set<String> includeAssetPatterns,
+            final Set<String> excludeFolderPatterns,
+            final Set<String> excludeAssetPatterns
+    ) {
+
+        if (path == null || path.isEmpty()) {
```

Comment:
you can encapsulate all these validation on a method, and just return a bean or tuple with the site and folderPath

Reply:
Done